src/watershed.jl

```diff
@@ -1,14 +1,18 @@
 import Base.isless
 
-struct PixelKey{CT}
-    val::CT
+struct PixelKey{N}
+    val
```

Comment:
Removing the type-parameter here seems likely to cause a performance hit: the priority queue needs to order its items, and the `isless` method accesses `val`. What was the motivation for removing it rather than just adding a second parameter `N`? Was it causing problems? If so can we address this in another way?